### PR TITLE
[feature] parse user algorithm

### PR DESCRIPTION
**Pull request by martinms-usc**

### CODE REVIEW

File: server/utilities.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+var queries = require('./queries.js');
+
+module.exports.evalAlg = function(userInput, dataType) {
+	// TODO: add queries file, get data inputs from database, save in respective variables
+	var data = queries.getSets(dataType);
+
+	/**
```
Comment (ghost) [inline server/utilities.js]: There should be one \* following the slash.

File: server/utilities.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	var pow2 = runTimeAverage(userInput, data[0], 8);
+	var pow3 = runTimeAverage(userInput, data[1], 6);
+	var pow4 = runTimeAverage(userInput, data[2], 3);
+	var pow5 = getRunTime(userInput, data[3]);
+	var pow6 = getRunTime(userInput, data[4]);
+	// var pow7 = getRunTime(userInput, data[5]);
+
+	return [pow2, pow3, pow4, pow5, pow6];
+}
+
```
Comment (mmurkidjanian) [inline server/utilities.js]: remove this line so there is only one line between each function declaration

File: server/utilities.js (hunk 1 above)
Comment (ghost) [inline server/utilities.js]: Also, please convert tabs to spaces for all of your indentation. An indent should be two spaces.

File: server/utilities.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+
+// TODO: transfer this to be method on codemirror class(?)
+function getFuncName(string) {
+  var dStop = string.indexOf('=');
+  var eStop = string.indexOf('(');
+  var funcName;
+  if (string.substr(0,3) === 'var') {
```
Comment (mmurkidjanian) [inline server/utilities.js]: insert line above if statement

File: server/utilities.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  return userAlg;
+}
+
+
+
+// TODO: transfer this to be method on codemirror class(?)
+function getFuncName(string) {
+  var dStop = string.indexOf('=');
+  var eStop = string.indexOf('(');
+  var funcName;
```
Comment (ghost) [inline server/utilities.js:98]: insert a blank line beneath your last variable declaration

File: server/utilities.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  } else {
+    return [Number(runTime.toFixed(3)), dbInput.length, null];
+  }
+}
+
+function buildFunc(userInput) {
+  var param = userInput.slice(userInput.indexOf('(') + 1, userInput.indexOf(')'));
+  var algString = userInput.slice(userInput.indexOf('{') + 1, userInput.lastIndexOf('}'));
+  var userAlg = new Function(param, algString);
+  console.log('U67-created algorithm with user input');
```
Comment (ghost) [inline server/utilities.js:87]: insert blank line beneath your last variable declaration

File: server/utilities.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  return coords;
+}
+
+function runTimeAverage(userInput, dbInput, iterations) {
+  console.log('U34-calculating runtime average for N=' + dbInput.length);
+  var times = [];
+  var total = 0;
+  var i = 0;
+  var averageRun;
+  var result;
```
Comment (ghost) [inline server/utilities.js:50]: insert blank line beneath your last variable declaration

File: server/utilities.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  var pow5 = getRunTime(userInput, data[3]);
+  var pow6 = getRunTime(userInput, data[4]);
+  // var pow7 = getRunTime(userInput, data[5]);
+
+  return [pow2, pow3, pow4, pow5, pow6];
+}
+
+module.exports.getCoords = function(data) {
+  console.log('U24-getting graph coordinates from evaluation data');
+  var coords = [];
```
Comment (ghost) [inline server/utilities.js:29]: insert blank line beneath this variable declaration

File: server/utilities.js [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+  var pow4 = runTimeAverage(userInput, data[2], 3);
+  var pow5 = getRunTime(userInput, data[3]);
+  var pow6 = getRunTime(userInput, data[4]);
+  // var pow7 = getRunTime(userInput, data[5]);
+
+  return [pow2, pow3, pow4, pow5, pow6];
+}
+
+module.exports.getCoords = function(data) {
+  console.log('U24-getting graph coordinates from evaluation data');
```
Comment (ghost) [inline server/utilities.js:27]: insert a line beneath this console.log

File: server/middleware.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+var utils = require('./utilities.js');
+
+module.exports.evalForAllInputSizes = function(req, res, next) {
+  var userInput = req.body.userInput;
+  console.log('M6-string received from ajax: ' + userInput);
+  var dataType = req.params.dataType || null;
+  var algorithmData = utils.evalAlg(userInput, dataType);
+  var d3Coordinates = utils.getCoords(algorithmData);
```
Comment (ghost) [inline server/middleware.js:10]: insert a line beneath the last variable declaration

File: server/middleware.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+var utils = require('./utilities.js');
+
+module.exports.evalForAllInputSizes = function(req, res, next) {
+  var userInput = req.body.userInput;
+  console.log('M6-string received from ajax: ' + userInput);
```
Comment (ghost) [inline server/middleware.js:6]: insert a line beneath this console.log

**Pull request merged**